pkg/pipeline/resource_pipeline_artifactory_integration.go

```diff
@@ -0,0 +1,138 @@
+package pipeline
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func pipelineArtifactoryProjectIntegrationResource() *schema.Resource {
+
+	var artifactoryIntegrationSchema = util.MergeSchema(
+		projectIntegrationSchema,
+		map[string]*schema.Schema{
+			"apikey": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "ApiKey for Artifactory access",
+			},
+			"user": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "User for Artifactory access",
+			},
+			"url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "URL to Artifactory instance",
+			},
+			"master_integration_id": {
+				Type:        schema.TypeInt,
+				Default:     98,
+				Optional:    true,
+				Description: "The Id of the master integration.",
+			},
+			"master_integration_name": {
```

Comment:
I assume this attribute is here (vs `projectIntegrationSchema`) is so you can set resource specific default value?

Reply:
refactored the id and name of integrations to the shared schema